Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -135,7 +149,9 @@ - (dispatch_queue_t)methodQueue
 
 - (void)beginLocationUpdatesWithDesiredAccuracy:(CLLocationAccuracy)desiredAccuracy distanceFilter:(CLLocationDistance)distanceFilter
 {
-  [self requestAuthorization];
+  if (!_locationConfiguration.skipPermissionRequests) {
```

Comment:
@chirag04 That's an interesting suggestion. A downside to that design is it'll open the door to a tricky to catch app bug. Suppose the app wants to request permissions on its own but fails to. With the current implementation, the app will see an exception due to not having the proper permissions. With your suggestion, RN will request permissions on behalf of the app making it difficult for the developer to realize there's a bug in their app.